### PR TITLE
Feat/RTENU-183 delete file

### PR DESCRIPTION
**Pull request by saneDG**
User can select file from list. Selected item has blue background color.
image

User can see selected file in modal before request is invoked. User can cancel or delete selected file.
image

Deleted file is removed from the file list.

This is supposed to work as minimum implementation of file delete feature. UI **will** be fixed in the future.

- New components: [FileDeleteDialog.tsx](https://github.com/finnishtransportagency/ratatiedot-extranet/compare/feat/RTENU-183-delete-file?expand=1#diff-c4ac3fdb2209d0dac0a447d5ce50c273d16c7bfef6c989242be4db3dcbebaa2d), [FileDeleteDialogButton.tsx](https://github.com/finnishtransportagency/ratatiedot-extranet/compare/feat/RTENU-183-delete-file?expand=1#diff-aadd9d2e8ed017c9d9fc7bda014250e4f12c18fb92d7155271c42ff760a933ee)
- 
- New functions
  - deleteFile: Delete sigle file

### CODE REVIEW

**Review by NinaDang97 — CHANGES_REQUESTED** [review]
Overall LGTM! But same question: could we delete file and the file list updates new change (e.g. by removing the successfully deleted file)? Currently, the file list remains the same after I deleted one file

**Comment by saneDG** [comment]
Updated PR:
- Allow deleting only one file at once
- Update list after file has been deleted
- Update translations. Necessary due to the changes mentioned above.

**Comment by saneDG** [comment]
Selected item background color might be too subtle to even see in some monitors. I would like do some changes to that later.

File: packages/frontend/src/components/Files/CategoryFiles.tsx [hunk 1]
```diff
@@ -74,18 +77,30 @@ export const CategoryFiles = ({ categoryName }: TCategoryFilesProps) => {
     return selectedFile?.entry.id === node.entry.id;
   };
 
+  const deleteDile = (node: TNode) => {
+    // find index of node and remove it from array
```
Comment (NinaDang97) [inline packages/frontend/src/components/Files/CategoryFiles.tsx:81]: deleteFile? :D

File: packages/frontend/src/components/Files/CategoryFiles.tsx [hunk 2]
```diff
@@ -56,12 +65,51 @@ export const CategoryFiles = ({ categoryName }: TCategoryFilesProps) => {
...
+  const isSelected = (node: TNode) => {
+    return selectedFile?.entry.id === node.entry.id;
+  };
+
+  const deleteDile = (node: TNode) => {
+    // find index of node and remove it from array
+    const index = fileList.findIndex((f) => f.entry.id === node.entry.id);
+    if (index > -1) {
+      const newList = [...fileList];
+      newList.splice(index, 1);
```
Comment (NinaDang97) [inline packages/frontend/src/components/Files/CategoryFiles.tsx:85]: Or instead can use `fileList.slice(index, 1)` so we won't need `newList`?

Comment (NinaDang97) [reply packages/frontend/src/components/Files/CategoryFiles.tsx:85]: And `totalFiles - 1` as well

**Review by NinaDang97 — APPROVED** [review]
Remember to decrease `totalFiles` after deletion, and we are good to go!

File: packages/frontend/src/components/Files/StaticFileCard.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import Image from '../../assets/icons/Image.svg';
+import PPT from '../../assets/icons/PowerPoint.svg';
+import { Colors } from '../../constants/Colors';
+import { DateFormat } from '../../constants/Formats';
+import { getLocaleByteUnit } from '../../utils/helpers';
+import { LocaleLang } from '../../constants/Units';
+import { TNode } from '../../types/types';
+import { Styles } from '../../constants/Styles';
+
+const NodeTypes = {
```
Comment (NinaDang97) [inline packages/frontend/src/components/Files/StaticFileCard.tsx:20]: NodeTypes is duplicated in File.tsx, can export

File: packages/frontend/src/components/Files/CategoryFiles.tsx [hunk 4]
```diff
@@ -36,7 +36,9 @@ export const CategoryFiles = ({ categoryName }: TCategoryFilesProps) => {
   const getCategoryFiles = useCallback(async () => {
     try {
       setLoading(true);
-      const response = await axios.get(`/api/alfresco/files?category=${getRouterName(categoryName)}&page=${page}`);
+      const response = await axios.get(
+        `http://localhost:3002/api/alfresco/files?category=${getRouterName(categoryName)}&page=${page}`,
```
Comment (NinaDang97) [inline packages/frontend/src/components/Files/CategoryFiles.tsx:40]: Could you remove localhost?

**Pull request merged**